### PR TITLE
fix: show correct total gpu capacity [DET-3733]

### PR DESCRIPTION
**Pull request by EmilyBonar**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by hkang1 — COMMENTED** [review]
The code looks good overall. There are some weird cases with k8s but I don't think we need to address them in this PR as we have some additional details to iron out. One improvement suggestion and one question.

File: webui/react/src/pages/Cluster/ClusterOverview.tsx [hunk 1]
```diff
@@ -82,6 +82,27 @@ const ClusterOverview: React.FC = () => {
...
+      }, 0);
+  }, [ resourcePools ]);
+
+  const cpuTotalSlots = useMemo(() => {
+    return resourcePools.filter(pool => pool.slotType === ResourceType.CPU)
+      .reduce((totalSlots, currentPool) => {
+        return totalSlots + currentPool.maxAgents * (currentPool.slotsPerAgent ?? 0);
+      }, 0);
+  }, [ resourcePools ]);
+
```
[the remark this answers is not in the document]
Comment (EmilyBonar) [reply webui/react/src/pages/Cluster/ClusterOverview.tsx:105]: Makes sense!

**Comment by hkang1** [comment]
does this also impact the nav tabbar?

File: webui/react/src/pages/Cluster/ClusterOverview.tsx [hunk 2]
```diff
@@ -82,6 +82,28 @@ const ClusterOverview: React.FC = () => {
...
+          break;
+        case ResourceType.ROCM:
+          index = 1;
+          break;
+        case ResourceType.CPU:
+          index = 2;
+          break;
+        default:
+          index = undefined;
+      }
```
Comment (hkang1) [inline webui/react/src/pages/Cluster/ClusterOverview.tsx:100]: This works and you don't have to change this. An alternative is to create a Record map:
```
const resourceTypeIndexMap = {
  [ResourceType.CUDA]: 0,
  [ResourceType.ROCM]: 1,
  [ResourceType.CPU]: 2,
};
const index = resourceTypeIndexMap[pool.slotType];
```

**Review by hkang1 — APPROVED** [review]
LGTM!

File: webui/react/src/pages/Cluster/ClusterOverview.tsx (hunk 2 above)
Comment (EmilyBonar) [reply webui/react/src/pages/Cluster/ClusterOverview.tsx:100]: I'll keep that in mind!

**Pull request merged**